met-web/src/components/survey/create/CloneOptions.tsx

```diff
@@ -0,0 +1,232 @@
+    const [selectedEngagement, setSelectedEngagement] = useState<Engagement | null>(null);
+    const [loadingSurveys, setLoadingSurveys] = useState(true);
+    const [loadingEngagements, setLoadingEngagements] = useState(true);
+    const [isSaving, setIsSaving] = useState(false);
+
+    const {
+        surveyForm,
+        handleSurveyFormChange,
+        availableSurveys,
+        setAvailableSurveys,
+        availableEngagements,
+        setAvailableEngagements,
+    } = useContext(CreateSurveyContext);
+
+    const initialFormError = {
+        name: false,
+    };
+    const [formError, setFormError] = useState(initialFormError);
+
+    const handleChange = (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+        handleSurveyFormChange({
+            ...surveyForm,
+            [e.target.name]: e.target.value,
+        });
+
+        if (hasKey(formError, e.target.name) && formError[e.target.name]) {
+            setFormError({
+                ...formError,
+                [e.target.name]: false,
+            });
+        }
+    };
+
+    const handleFetchSurveys = async (
+        page: number,
+        size: number,
+        sort_order: 'asc' | 'desc' | undefined,
+        search_text?: string,
+    ) => {
+        try {
+            const fetchedSurveys = await getSurveysPage({
+                page: page,
+                size: size,
+                sort_order: sort_order,
+                search_text: search_text,
+            });
+            setAvailableSurveys(fetchedSurveys.items);
+            setLoadingSurveys(false);
+        } catch (error) {
+            dispatch(openNotification({ severity: 'error', text: 'Error occurred while fetching available surveys' }));
+        }
+    };
+
+    const handleFetchEngagements = async (
+        page: number,
+        size: number,
+        sort_order: 'asc' | 'desc' | undefined,
+        search_text?: string,
+    ) => {
+        try {
+            const fetchedEngagements = await getEngagements({
+                page: page,
+                size: size,
+                sort_order: sort_order,
+                search_text: search_text,
+            });
+            console.log('FETCH!');
+            console.log(fetchedEngagements);
+            setAvailableEngagements(
+                fetchedEngagements.items.filter((engagement, index) => engagement.surveys.length !== 0),
+            );
+            setLoadingEngagements(false);
+        } catch (error) {
+            dispatch(
+                openNotification({ severity: 'error', text: 'Error occurred while fetching available engagements' }),
+            );
+        }
+    };
+
+    useEffect(() => {
+        if (!availableSurveys) {
+            handleFetchSurveys(1, 10, 'asc', '');
+        }
+    }, [availableSurveys]);
+
+    useEffect(() => {
+        if (!availableEngagements) {
+            handleFetchEngagements(1, 10, 'asc', '');
```

Comment:
what if there's more than 10?

Reply:
ill set it up for fetching 2000 i was trying to make it so that we fetch 10 if user cant find it they type in the search bar and we fetch what we need

Reply:
Yup, figured you were going for that. Great idea. I'm thinking we can extend it and implement something like also when user scrolls and reaches end of list another page of items will be loaded. We just need to figure how to do that, seems doable and all, just feels a bit tricky